devtools/x/src/cargo.rs

```diff
@@ -15,15 +19,20 @@ use std::{
 const RUST_TOOLCHAIN_VERSION: &str = include_str!("../../../rust-toolchain");
 const RUSTUP_TOOLCHAIN: &str = "RUSTUP_TOOLCHAIN";
 const CARGO: &str = "CARGO";
+const SECRET_ENVS: &[&str] = &["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"];
 
 pub struct Cargo {
     inner: Command,
     pass_through_args: Vec<OsString>,
-    env_additions: Vec<(OsString, OsString)>,
+    env_additions: Vec<(OsString, Option<OsString>)>,
```

Comment:
I think the right data structure is a BTreeMap for this, not a Vec.

Reply:
It would need to be a BTreeMap<OsString, Option<OsString>>, so I could remove environment variables that are set to None.

Is that what you are expecting @sunshowers ?

Reply:
Yep :)

Reply:
Do you want to print out the None/Some status of secret env vars? 

I couldn't think of a good way to textually represent a removal of a variable....   I suppose I could output the bash commands directly.....
```
export FOO="blah"
unset BIZ
```

Reply:
So I went off and started converting this, but I guess I disagree.   I want to retain the order of the envs are added (not sort them), and don't mind seeing an env added twice (and override) if the code generates it, a user would see a slightly non-sensical double addition of an environment variable.   But as you would expect the last one written wins.

The order is  a useful hint to the flow.   Having ordering matters in places where grcov/sccache are both run -- or other tools that may need env flags set.    Ideally I'd have a data type that retained order of insertion, but removed any overwritten value and appended it at the end of the sequence and removing it from the prior location......   

I think a linked list backed order preserving set would be the ideal abstraction.   I wouldn't mind making this a linked list instead and performing the removes myself it you're okay with that.

Or we can leave it as is until we hit that situation?

Reply:
I like the idea of printing out the shell commands, makes it easier to copy-paste as well.

re the map, https://docs.rs/indexmap/1.6.0/indexmap/ may work as we discussed.

Reply:
done. thanks for the pointer.